core/src/test/java/org/elasticsearch/search/aggregations/metrics/scripted/InternalScriptedMetricTests.java

```diff
@@ -105,4 +151,52 @@ protected void assertReduced(InternalScriptedMetric reduced, List<InternalScript
         return InternalScriptedMetric::new;
     }
 
+    @Override
+    protected void assertFromXContent(InternalScriptedMetric aggregation, ParsedAggregation parsedAggregation) {
+        assertTrue(parsedAggregation instanceof ParsedScriptedMetric);
+        ParsedScriptedMetric parsed = (ParsedScriptedMetric) parsedAggregation;
+
+        assertValues(aggregation.aggregation(), parsed.aggregation());
+    }
+
+    private static void assertValues(Object expected, Object actual) {
+        if (expected instanceof Long) {
+            // longs that fit into the integer range are parsed back as integer
+            if (actual instanceof Integer) {
+                assertEquals(((Long) expected).intValue(), actual);
+            } else {
+                assertEquals(expected, actual);
+            }
+        } else if (expected instanceof Float) {
+            // based on the xContent type, floats are sometimes parsed back as doubles
+            if (actual instanceof Double) {
+                assertEquals(expected, ((Double) actual).floatValue());
+            } else {
+                assertEquals(expected, actual);
+            }
+        } else if (expected instanceof GeoPoint) {
+            assertTrue(actual instanceof Map);
+            GeoPoint point = (GeoPoint) expected;
+            Map<String, Object> pointMap = (Map<String, Object>) actual;
+            assertEquals(point.getLat(), pointMap.get("lat"));
+            assertEquals(point.getLon(), pointMap.get("lon"));
```

Comment:
So here we have a difference between transport client and our rest client. Would it make sense to add an extra post-parsing treatment for maps that in case of a map of 2 keys lat and lon converts it back to a GeoPoint?